plugins/org.jkiss.dbeaver.ext.cubrid/src/org/jkiss/dbeaver/ext/cubrid/model/meta/CubridMetaModel.java

```diff
@@ -53,100 +65,362 @@ public String getTableOrViewName(GenericTableBase base)
+        return new CubridSynonym(container, name, description, dbResult);
+    }
+
+    @NotNull
+    @Override
+    public boolean supportsDatabaseTriggers(@NotNull GenericDataSource dataSource) {
+        return true;
+    }
+
+    @NotNull
+    @Override
+    public boolean supportsTriggers(@NotNull GenericDataSource dataSource) {
+        return true;
+    }
+
+    @NotNull
+    @Override
+    public JDBCStatement prepareTableTriggersLoadStatement(
+            @NotNull JDBCSession session,
+            @NotNull GenericStructContainer container,
+            @Nullable GenericTableBase table)
+            throws SQLException {
+        String sql = "select t1.*, t2.*, owner.name from db_trigger t1 join db_trig t2 \n"
+                + "on t1.name = t2.trigger_name where owner.name = ? \n"
+                + (table != null ? "and target_class_name = ?" : "");
+        final JDBCPreparedStatement dbStat = session.prepareStatement(sql);
+        dbStat.setString(1, container.getName());
+        if (table != null) {
+            dbStat.setString(2, table.getName());
+        }
+        return dbStat;
+    }
+
+    @Nullable
+    @Override
+    public CubridTrigger createTableTriggerImpl(
+            @NotNull JDBCSession session,
+            @NotNull GenericStructContainer container,
+            @Nullable GenericTableBase table,
+            @Nullable String triggerName,
+            @NotNull JDBCResultSet dbResult)
+            throws DBException {
+        String name = JDBCUtils.safeGetString(dbResult, "name");
+        String description = JDBCUtils.safeGetString(dbResult, "comment");
+        return new CubridTrigger(table, name, description, dbResult);
+    }
+
+    @NotNull
+    @Override
+    public JDBCStatement prepareContainerTriggersLoadStatement(
+            @NotNull JDBCSession session,
+            @NotNull GenericStructContainer container)
+            throws SQLException {
+        String sql = "select t1.*, t2.*, owner.name from db_trigger t1 join db_trig t2 \n"
+                + "on t1.name = t2.trigger_name where owner.name = ? \n";
+        final JDBCPreparedStatement dbStat = session.prepareStatement(sql);
+        dbStat.setString(1, container.getName());
+        return dbStat;
+    }
+
+    @Nullable
+    @Override
+    public CubridTrigger createContainerTriggerImpl(
+            @NotNull GenericStructContainer container,
+            @NotNull JDBCResultSet dbResult)
+            throws DBException {
+        String name = JDBCUtils.safeGetString(dbResult, "name");
+        String description = JDBCUtils.safeGetString(dbResult, "comment");
+        String tableName = JDBCUtils.safeGetString(dbResult, "target_class_name");
+        String owner = JDBCUtils.safeGetString(dbResult, "target_owner_name");
+        DBRProgressMonitor monitor = dbResult.getSession().getProgressMonitor();
+        CubridTable cubridTable = (CubridTable) container.getDataSource().findTable(monitor, null, owner, tableName);

```

Comment:
But way do you set null here instead of container?

Reply:
I cannot set container as an argument here because the parameter is a String type. And Cubrid doesn't have catalog name. So, what is your suggestion?